components/outputs/plot-altair/index.qmd

```diff
@@ -0,0 +1,220 @@
+    apps:
+    - title: Preview
+      file: app-variation-selection-interval-brush-express.py
+      height: 720
+    - title: Express
+      file: app-variation-selection-interval-brush-express.py
+      shinylive:
+      height: 720
+    - title: Core
+      file: app-variation-selection-interval-brush-core.py
+      shinylive:
+      height: 720
+---
+
+:::{#example}
+:::
+
+:::{#relevant-functions}
+:::
+
+## Details
+
+[Vega-Altair](https://altair-viz.github.io/index.html) is an interactive declarative visualization library in Python.
+It is based on the [Vega-Lite](https://vega.github.io/vega-lite/) grammar of interactive graphics.
+
+To make an interactive plot with Altair in Shiny for Python,
+we will need to use the [`shinywidgets`](https://shiny.posit.co/py/docs/jupyter-widgets.html)
+library to connect Shiny with [`ipywidgets`](https://ipywidgets.readthedocs.io/en/latest/)
+
+To make an Altair figure, we need to do the following steps:
+
+::: {.panel-tabset group="syntax"}
+## Express
+
+1. Import the `render_altair()` functions from `shinywidgets`.
+
+2. Define a function where you want to create the figure.
+    - If your function calls reactive values, Shiny will update your figure whenever those values change, in a [reactive fashion](https://shiny.posit.co/py/docs/reactive-programming.html).
+    - Do not forget to return the plot object from your function.
+
+3. Decorate your plotting function with a `@render_altair()` decorator.
+    - If your plotting function is not the same as the `id` you used in the `ui.output_widget()`, you can add an additional `@output(id=...)` decorator.
+
+## Core
+
+1. Import the `output_widget()` and `render_altair()` functions from `shinywidgets`.
+
+2. Call `output_widget()` to the UI of your app to create a div in which to display the figure.
+   Where you call this function will determine where the figure will appear within the layout of the app.
+   The `id` parameter you provide will be used to link to other parts of the Shiny app.
+
+3. Define a function within the `server()` function that creates the figure.
+    - The name of the function should be the same value you passed into the `id` parameter in your `output_widget()` function call in the UI.
+    - If your function calls reactive values, Shiny will update your figure whenever those values change, in a [reactive fashion](https://shiny.posit.co/py/docs/reactive-programming.html).
+    - Do not forget to return the plot object from your function.
+
+4. Decorate your plotting function with a `@render_altair()` decorator.
+    - If your plotting function is not the same as the `id` you used in the `ui.output_widget()`, you can add an additional `@output(id=...)` decorator.
+:::
+
+To learn more about using Jupyter widgets with Shiny visit: [shiny.posit.co/py/docs/jupyter-widgets.html](https://shiny.posit.co/py/docs/jupyter-widgets.html).
```

Comment:
I feel like this should come towards the end of the article (and mention that altair is just example of a Jupyter widget in Shiny)?

Reply:
Oh, now I see we do this for Plotly -- I guess it's not super important, but ideally we'd have something better on both pages. 

Related to this, a stretch goal (for both components) would be to have a few more "basic" variations before diving into the more complex "plots as inputs" and "plot mutation" concepts. 

I also think a link to plotly/atair galleries at this point might be more useful than a link to the generic Jupyter Widgets article.